lib/ansiblelint/runner.py

```diff
@@ -101,9 +87,25 @@ def run(self) -> List:
                 "Examining %s of type %s",
                 ansiblelint.file_utils.normpath(file['path']),
                 file['type'])
-            matches.extend(self.rules.run(file, tags=set(self.tags),
-                           skip_list=self.skip_list))
+            matches = matches.union(
+                self.rules.run(file, tags=set(self.tags),
+                               skip_list=self.skip_list))
         # update list of checked files
         self.checked_files.update([x['path'] for x in files])
 
-        return sorted(set(matches))
+        return sorted(matches)
+
+    def _yield_matches(self, files: List) -> Generator[MatchError, None, None]:
```

Comment:
The name should be made better. yield matches only answered the question about the return type not providing any details oh how it does that, based on what etc. "yield" is an implementation detail that doesn't really improve the semantical reference...

P.S. I usually prefer using "emit" to signal that the function is a generator.

Reply:
It seems like this doesn't run any checks except for loading-related ones. This could be reflected too.

Reply:
sure.